### PR TITLE
gThumb 3.12.4

### PR DESCRIPTION
**Pull request by fedelibre**

### CODE REVIEW

**Comment by hfiguiere** [comment]
Why asking me for a review? I don't maintain that package.

**Comment by fedelibre** [comment]
Because you contributed a few commits in the past.

There isn't much to review, except the removed `--filesystem=xdg-config/gtk-3.0` permission.

**Comment by AsciiWolf** [comment]
Please, consider also updating the runtime to 45 as a part of this PR. Thanks!

File: org.gnome.gThumb.yml [hunk 1]
```diff
@@ -79,8 +79,8 @@ modules:
             url: https://github.com/strukturag/libde265/releases/download/v1.0.8/libde265-1.0.8.tar.gz
             sha256: 24c791dd334fa521762320ff54f0febfd3c09fc978880a8c5fbc40a88f21d905
     sources:
-      - url: https://github.com/strukturag/libheif/releases/download/v1.12.0/libheif-1.12.0.tar.gz
-        sha256: e1ac2abb354fdc8ccdca71363ebad7503ad731c84022cf460837f0839e171718
+      - url: https://github.com/strukturag/libheif/releases/download/v1.17.6/libheif-1.17.6.tar.gz
+        sha256: 8390baf4913eda0a183e132cec62b875fb2ef507ced5ddddc98dfd2f17780aee
         type: archive
 
   - name: libjxl
```
Comment (hfiguiere) [inline org.gnome.gThumb.yml:86]: libjxl 0.8.2 is in the runtime now.

Comment (hfiguiere) [reply org.gnome.gThumb.yml:86]: since it's the build failure.

File: org.gnome.gThumb.yml [hunk 2]
```diff
@@ -144,14 +122,11 @@ modules:
       - install -Dm644 ../data/appdata/org.gnome.gThumb.appdata.xml.in $FLATPAK_DEST/share/appdata/$FLATPAK_ID.appdata.xml
```
Comment (hfiguiere) [inline org.gnome.gThumb.yml:122]:
```
Exporting share/appdata/org.gnome.gThumb.appdata.xml
Exporting share/metainfo/org.gnome.gThumb.appdata.xml
```

there is defintely one too many

Upstream does:
```
Installing data/appdata/org.gnome.gThumb.appdata.xml to /app/share/metainfo
```
so this should be removed here.

File: org.gnome.gThumb.yml (hunk 1 above)
Comment (fedelibre) [reply org.gnome.gThumb.yml]: Thanks, removed

**Comment by hfiguiere** [comment]
Note that:
```
Run-time dependency libsoup-2.4 found: NO (tried pkgconfig and cmake)
Run-time dependency json-glib-1.0 found: YES 1.8.0
Run-time dependency webkit2gtk-4.0 found: NO (tried pkgconfig and cmake)
```

`libsoup-2.4` has been removed from the runtime but is available as shared-module.
`webkit2gtk-4.0` is no longer because it requires libsoup-2.4, and is replaced by `webkit2gtk-4.1` that uses libsoup-3.0.

**Comment by hfiguiere** [comment]
that might be a problem.

**Comment by fedelibre** [comment]
Optional libraries:
> libsoup, json-glib, webkit2gtk, libsecret - upload images to and
download images from some web services such as Facebook, Flickr

I might install libsoup-2.4 from shared-modules and add webkit2gtk-4.0 to the manifest: here's [an example](https://github.com/flathub/in.cinny.Cinny/blob/263ff5d11ee96368a36e24e5adffc742f1b93f2d/in.cinny.Cinny.yml#L32).

**Comment by fedelibre** [comment]
I can't understand why it's failing. The path to the shared module is correct.

**Comment by hfiguiere** [comment]
not in the version of the git module that is set in this repository. You need to update it.

**Pull request merged**